src/main/java/core/basesyntax/Ball.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Ball {
+    public final Colors color;
+    public final int number;
+    private final ColorSupplier colorSupply = new ColorSupplier();
+
+    Ball(int number) {
+        this.color = colorSupply.getRandomColor();
```

Comment:
move this logic into getRandomBall() method

and add access modifier to constructor